pakyow-presenter/lib/pakyow/presenter/form.rb

```diff
@@ -0,0 +1,65 @@
+module Pakyow
+  module Presenter
+    class Form
+
+      attr_reader :form_view, :context
+
+      def initialize(view, context)
+        @form_view = view
+        @context = context
+      end
+
+      def create(*binding_args, &block)
+        bind_data(binding_args, block) do |routes, view|
+          view.attrs.action = routes.path(:create, request_params) if routes
+        end
+      end
+
+      def update(*binding_args, &block)
+        bind_data(binding_args, block) do |routes, view|
+          if routes
+            route_params = request_params.merge(
+              :"#{scoped_as}_id" => view.attrs.__send__('data-id').value
+            )
+
+            view.attrs.action = routes.path(:update, route_params)
+          end
+
+          view.prepend(
+            View.new('<input type="hidden" name="_method" value="patch">')
+          )
+        end
+      end
+
+      private
+
+      def bind_data(binding_args, block)
+        bind(*binding_args) do |view, data|
+          routes = Router.instance.group(scoped_as)
+
+          yield(routes, view)
+
+          return if block.nil?
+
+          if block.arity == 1
+            view.instance_exec(&block)
+          else
+            block.call(view, data)
+          end
+        end
+      end
+
+      def request_params
+        context.request.params
+      end
+
+      def method_missing(method, *args, &block)
```

Comment:
Yeah. My original idea was for form to act like a view, but with extra form actions, but since the user can access the view maybe that can be the solution to the problem above as well, so that the example above would simply be:

```ruby
data = { id: 1, name: 'a', images: [{ id: 1, name: 'b' }, { id: 2, name: 'c' }] }
form = view.form(:album)
form.update(data)
form.view.scope(:image).apply(data[:images]) do |fields, image|
  fields.prop(:name).attrs.name = "album[images][#{image[:id]}[name]"
end
```